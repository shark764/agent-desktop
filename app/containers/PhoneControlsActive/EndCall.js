/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { setInteractionConfirmation } from 'containers/AgentDesktop/actions';

import CircleIconButton from 'components/CircleIconButton';
import messages from './messages';

const styles = {
  circleIconButtonRow: {
    padding: '0 1.5px',
  },
};

export class EndCall extends React.PureComponent {
  confirmEndInteraction = () => {
    if (this.props.interactionStatusIsFatal) {
      CxEngage.interactions.end({
        interactionId: this.props.interactionId,
      });
    } else {
      this.props.setInteractionConfirmation(this.props.interactionId, true);
    }
  };

  render() {
    return (
      <CircleIconButton
        id="endCallButton"
        name="endCall"
        title={messages.endCall}
        onClick={this.confirmEndInteraction}
        style={styles.circleIconButtonRow}
      />
    );
  }
}

EndCall.propTypes = {
  interactionId: PropTypes.string.isRequired,
  interactionStatusIsFatal: PropTypes.bool.isRequired,
  setInteractionConfirmation: PropTypes.func.isRequired,
};

function mapDispatchToProps(dispatch) {
  return {
    setInteractionConfirmation: (interactionId, status) =>
      dispatch(setInteractionConfirmation(interactionId, status)),
    dispatch,
  };
}

export default connect(
  null,
  mapDispatchToProps
)(EndCall);
