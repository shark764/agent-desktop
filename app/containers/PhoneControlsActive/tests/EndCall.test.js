/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';

import { EndCall } from '../EndCall';

describe('<EndCall />', () => {
  it('renders correctly', () => {
    const rendered = shallow(
      <EndCall
        interactionId="mock-interaction-id"
        interactionStatusIsFatal
        setInteractionConfirmation={() => {}}
      />
    );
    expect(rendered).toMatchSnapshot();
  });

  describe('when interactionStatusIsFatal', () => {
    beforeEach(() => {
      global.CxEngage = {
        interactions: {
          end: jest.fn(),
        },
      };
    });
    const rendered = shallow(
      <EndCall
        interactionId="mock-interaction-id"
        interactionStatusIsFatal
        setInteractionConfirmation={() => {}}
      />
    );
    it('calls CxEngage.interactions.end when clicking on icon', () => {
      rendered.find('#endCallButton').simulate('click');
      expect(global.CxEngage.interactions.end.mock.calls.length).toBe(1);
      expect(global.CxEngage.interactions.end.mock.calls[0][0]).toEqual({
        interactionId: 'mock-interaction-id',
      });
    });
  });

  describe('when interactionStatusIsFatal is false', () => {
    const setInteractionConfirmation = jest.fn();
    const selectInteraction = jest.fn();
    const rendered = shallow(
      <EndCall
        interactionId="mock-interaction-id"
        interactionStatusIsFatal={false}
        setInteractionConfirmation={setInteractionConfirmation}
        selectInteraction={selectInteraction}
      />
    );
    it('calls setInteractionConfirmation when clicking on icon', () => {
      rendered.find('#endCallButton').simulate('click');
      expect(selectInteraction.mock.calls[0][0]).toBe('mock-interaction-id');
      expect(setInteractionConfirmation.mock.calls.length).toBe(1);
      expect(setInteractionConfirmation.mock.calls[0][0]).toBe(
        'mock-interaction-id'
      );
      expect(setInteractionConfirmation.mock.calls[0][1]).toBe(true);
    });
  });
});
