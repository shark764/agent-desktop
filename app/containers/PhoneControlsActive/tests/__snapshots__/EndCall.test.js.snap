// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EndCall /> renders correctly 1`] = `
<InjectIntl(CircleIconButton)
  id="endCallButton"
  name="endCall"
  onClick={[Function]}
  style={
    Object {
      "padding": "0 1.5px",
    }
  }
  title={
    Object {
      "defaultMessage": "End Call",
      "id": "app.containers.PhoneControls.endCall",
    }
  }
/>
`;
